src/main/java/com/google/sps/data/Room.java

```diff
@@ -14,20 +14,113 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.EmbeddedEntity;
 import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.DatastoreFailureException;
 import java.util.stream.Collectors;
 import java.util.Map;
 import java.util.stream.*;
+import com.google.appengine.api.datastore.KeyFactory;
 
+//Object repsresenting the Room that the chat and video streaming will be in
+//TODO: Restgructure so logic is more split up
 public class Room {
+    public static final int MAX_MESSAGES = 10;
     private static final String ROOM_ENTITY = "Room";
     private static final String MEMBERS_PROPERTY = "members";
     private static final String MESSAGES_PROPERTY = "messages";
     private static final String VIDEOS_PROPERTY = "videos";
     private static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     private List<Member> members;
-    private LinkedList<Message> messages;
+    private Queue<Message> messages;
     private Queue<Video> videos;
 
+    //Room factory function
+    public static Room createRoom(List<Member> members, Queue<Video> videos, Queue<Message> messages){
+        return new Room(members, videos, messages);
+    }
+
+    // Manipulates the messages property of the room given a room key and a new message to be added
+    public static void addMessagesFromKey(Key roomKey, Message chatMessage) {
+        try {
+        Entity roomEntity = datastore.get(roomKey);
+        Room room = Room.fromEntity(roomEntity);
+        room.addMessage(chatMessage);
+
+        roomEntity.setProperty("messages", room.getMessagesAsEntities());
+        datastore.put(roomEntity);
+        } catch (EntityNotFoundException e){
+            System.out.println(e.toString());
+        }
+    }
```

Comment:
Remove this function.

Reply:
Would it be better to have a function that returns a roomEntity? (I need the room entity -not the object- to update the messages property in datastore)